include/hipSYCL/sycl/libkernel/half.hpp

```diff
@@ -65,6 +68,10 @@ class half {
     return fp16::promote_to_double(_data);
   }
 
+  operator int() const {
+    return fp16::promote_to_int(_data);
```

Comment:
`return static_cast<int>(fp16::promote_to_float(_data))`?

I think turning a half into an int is not really a promotion, it's a conversion.